### PR TITLE
DDP-6297 allow prism admin to edit user activities

### PR DESCRIPTION
**Pull request by aglushko-broad**
If you're ok with the implemented approach I'll add unit tests for the admin editing panel
most of the removed code was added in terms of the PR https://github.com/broadinstitute/ddp-angular/pull/569

image
image

image
image

### CODE REVIEW

**Comment by aglushko-broad** [comment]
@MocanaAtBroad should we change the dashboard url approach from 
'dashboard/:userGuid' to 
'dashboard?user_guid=user_guid' 
It'd look more consistent

**Comment by MocanaAtBroad** [comment]
Do we need to carry the user guid in the URL? We keep it in local store, right?

**Comment by MocanaAtBroad** [comment]
When in Prism, when get redirected back to dashboard after hitting "Submit" at the end of the survey being edited, we don't show any activities. Perhaps we are losing the user guid being edited? (The user being edited is preserved in testboston).

**Comment by aglushko-broad** [comment]
> Do we need to carry the user guid in the URL? We keep it in local store, right?

No, actually we don't (I put participation user data into session storage for better performance in the previous commit, but removed it later since it's sensitive data). See the details below in the comment

**Comment by aglushko-broad** [comment]
> When in Prism, when get redirected back to dashboard after hitting "Submit" at the end of the survey being edited, we don't show any activities. Perhaps we are losing the user guid being edited? (The user being edited is preserved in testboston).

oops, you're right. The problem was that actions redirects use query param and initial dashboard prism implementation used url param. I changed it to queryParam too for now.
testboston works differently: it adds participantGuid into the session (locates in local storage)
how it works:
testboston (old prism, only one user per search): admin searches by invitation id -> finds the user -> add participantGuid of found user guid to the localstorage (and add invitation code) -> reset the session guid when admin changes search string
angio and brain (new prism, multiple users per search): admin searches by any string -> finds the list of users -> navigates through the dashboard and actions taking participantGuid from url query param (and requests additional participant data by participantGuid  from the url)

actually, I'm starting to think that there is no advantage of having this new url param strategy. I remember we discussed it in the chat with Yufeng before the implementation and you mentioned that you expected it would look like that:  /prism/patient/PATIENT_GUID and I also thought it might be a good idea to have a participant guid in the url param. But now I think it's probably fine to have participantGuid in the session for the new prism as well as it was done for testboston prism. It allows simpler reuse and removing of the logic we implemented for using selectedUserGuid from url param.

@MocanaAtBroad  What do you think?

**Comment by MocanaAtBroad** [comment]
> > When in Prism, when get redirected back to dashboard after hitting "Submit" at the end of the survey being edited, we don't show any activities. Perhaps we are losing the user guid being edited? (The user being edited is preserved in testboston).
> 
> oops, you're right. The problem was that actions redirects use query param and initial dashboard prism implementation used url param. I changed it to queryParam too for now.
> testboston works differently: it adds participantGuid into the session (locates in local storage)
> how it works:
> testboston (old prism, only one user per search): admin searches by invitation id -> finds the user -> add participantGuid of found user guid to the localstorage (and add invitation code) -> reset the session guid when admin changes search string
> angio and brain (new prism, multiple users per search): admin searches by any string -> finds the list of users -> navigates through the dashboard and actions taking participantGuid from url query param (and requests additional participant data by participantGuid from the url)
> 
> actually, I'm starting to think that there is no advantage of having this new url param strategy. I remember we discussed it in the chat with Yufeng before the implementation and you mentioned that you expected it would look like that: /prism/patient/PATIENT_GUID and I also thought it might be a good idea to have a participant guid in the url param. But now I think it's probably fine to have participantGuid in the session for the new prism as well as it was done for testboston prism. It allows simpler reuse and removing of the logic we implemented for using selectedUserGuid from url param.
> 
> @MocanaAtBroad What do you think?

When we GET activities, the URL used already includes the user guid, e.g.:
https://pepper-dev.datadonationplatform.org/pepper/v1/user/XXOWGMTY26BZ9IESDWI0*/studies/ANGIO/activities/VYFAC1HJN5

**XXOWGMTY26BZ9IESDWI0** is the user GUID, so redundant to also have it in a query param.

Perhaps it makes sense to have the GUID in URL going from the results to the dashboard? Maybe? If you think it makes sense go for it.

It looks like we store at least one result in storage after query, but we don't read it? (was looking at usage of SessionStorageService) . Could not see that we were reading it back but maybe I missed where. If we don't read it back we should get rid of it.

Otherwise, this is looking good and does everything that we need it to do as far as I can see.

File: ddp-workspace/projects/ddp-angio/src/app/app.module.ts [hunk 1]
```diff
@@ -96,65 +95,64 @@ config.prismRoute = 'prism';
...
+            const locale = 'en';
+            translate.setDefaultLang(locale);
+            translate.use(locale).subscribe(() => {
+                logger.logEvent(LOG_SOURCE, `Successfully initialized '${locale}' language as default.`);
+            }, err => {
+                logger.logError(LOG_SOURCE, `Problem with '${locale}' language initialization`, err);
+            }, () => {
+                resolve(null);
+            });
+        });
```
Comment (aglushko-broad) [inline ddp-workspace/projects/ddp-angio/src/app/app.module.ts:111]: only indent changes

File: ddp-workspace/projects/ddp-angio/src/app/app.module.ts [hunk 2]
```diff
@@ -96,65 +95,64 @@ config.prismRoute = 'prism';
...
-    DdpModule,
-    ToolkitModule
-  ],
-  declarations: [
-    AboutUsComponent,
-    DataReleaseComponent,
-    MoreDetailsComponent,
-    PrismComponent,
-    WelcomeComponent,
-    DashboardComponent
```
Comment (aglushko-broad) [inline ddp-workspace/projects/ddp-angio/src/app/app.module.ts:129]: DashboardComponent was removed, rest - indent

File: ddp-workspace/projects/ddp-brain/src/app/app.module.ts [hunk 3]
```diff
@@ -96,73 +95,72 @@ sdkConfig.lookupPageUrl = AppRoutes.Prism;
...
+            const locale = 'en';
+            translate.setDefaultLang(locale);
+            translate.use(locale).subscribe(() => {
+                logger.logEvent(LOG_SOURCE, `Successfully initialized '${locale}' language as default.`);
+            }, err => {
+                logger.logError(LOG_SOURCE, `Problem with '${locale}' language initialization:`, err);
+            }, () => {
+                resolve(null);
+            });
+        });
```
Comment (aglushko-broad) [inline ddp-workspace/projects/ddp-brain/src/app/app.module.ts:111]: only indent

File: ddp-workspace/projects/ddp-brain/src/app/app.module.ts [hunk 4]
```diff
@@ -96,73 +95,72 @@ sdkConfig.lookupPageUrl = AppRoutes.Prism;
...
-    WelcomeComponent,
-    HeaderComponent,
-    FooterComponent,
-    FaqComponent,
-    DataComponent,
-    AboutUsComponent,
-    ModalImageComponent,
-    LearnMoreSectionComponent,
-    PrismComponent,
-    DashboardComponent,
```
Comment (aglushko-broad) [inline ddp-workspace/projects/ddp-brain/src/app/app.module.ts:128]: DashboardComponent was removed, rest - indent

File: ddp-workspace/projects/ddp-sdk/src/lib/components/dashboard.component.ts [hunk 5]
```diff
@@ -1,36 +1,24 @@
...
     template: `
-    <h2 class="PageContent-subtitle PageContent-subtitle-dashboard" translate>
-        SDK.Dashboard.Title
-    </h2>
-    <p class="PageContent-text PageContent-text-dashboard" translate>
-        SDK.Dashboard.Text
-    </p>
-    <ddp-user-activities [studyGuid]="studyGuid"
-                         [selectedUserGuid]="selectedUserGuid"
-                         [displayedColumns]="activitiesColumns"
```
Comment (aglushko-broad) [inline ddp-workspace/projects/ddp-sdk/src/lib/components/dashboard.component.ts:15]: selectedUserGuid and displayedColumns were removed here only

File: ddp-workspace/projects/ddp-sdk/src/lib/components/prism/prism.component.ts [hunk 6]
```diff
@@ -14,140 +15,145 @@ import { SessionStorageService } from '../../services/sessionStorage.service';
...
-  @ViewChild(StickyScrollDirective, { static: true }) stickyScroll: StickyScrollDirective;
-
-  public searchField = new FormControl();
-  public displayedColumns: string[] = [];
-  public dataSource = new MatTableDataSource<SearchParticipant>([]);
-  public totalCount = 0;
-  public isSearchLoading = false;
-  public isSearchDebounce = false;
-  public MIN_SEARCH_LENGTH = 4;
-  public dashboardRoute: string;
```
Comment (aglushko-broad) [inline ddp-workspace/projects/ddp-sdk/src/lib/components/prism/prism.component.ts:33]: dashboardRoute was removed

File: ddp-workspace/projects/ddp-sdk/src/lib/components/prism/prism.component.ts [hunk 7]
```diff
@@ -14,140 +15,145 @@ import { SessionStorageService } from '../../services/sessionStorage.service';
...
+    private readonly searchQueryStorageName: string;
+    private readonly paginationSizeStorageName: string;
+    private readonly paginationIndexStorageName: string;
+    readonly paginationSizes = [10, 25, 100];
+
+    constructor(
+        private participantsSearch: ParticipantsSearchServiceAgent,
+        private router: Router,
+        private storageService: SessionStorageService,
+        private sessionService: SessionMementoService,
```
Comment (aglushko-broad) [inline ddp-workspace/projects/ddp-sdk/src/lib/components/prism/prism.component.ts:49]: SessionMementoService injected

File: ddp-workspace/projects/ddp-sdk/src/lib/components/prism/prism.component.ts [hunk 8]
```diff
@@ -14,140 +15,145 @@ import { SessionStorageService } from '../../services/sessionStorage.service';
...
+        @Inject('ddp.config') private config: ConfigurationService,
     ) {
-    this.displayedColumns = this.config.prismColumns;
-    this.dashboardRoute = this.config.prismDashboardRoute;
```
Comment (aglushko-broad) [inline ddp-workspace/projects/ddp-sdk/src/lib/components/prism/prism.component.ts:52]: dashboardRoute was removed

File: ddp-workspace/projects/ddp-sdk/src/lib/components/prism/prism.component.ts [hunk 9]
```diff
@@ -14,140 +15,145 @@ import { SessionStorageService } from '../../services/sessionStorage.service';
...
+        this.paginator.pageIndex = event.pageIndex;
+        this.paginator.pageSize = event.pageSize;
+        this.paginator.page.emit(event);
+    }
+
+    get isPaginationHidden(): boolean {
+        return this.isSearchLoading || this.dataSource.data.length <= this.paginationSizes[0];
+    }
+
+    public clickOnUser(user: SearchParticipant): void {
```
Comment (aglushko-broad) [inline ddp-workspace/projects/ddp-sdk/src/lib/components/prism/prism.component.ts:155]: clickOnUser method was added

**Review by MocanaAtBroad — DISMISSED** [review]
LGTM!

**Review by MocanaAtBroad — APPROVED** [review]

**Pull request merged**